modules/videoio/test/test_ffmpeg.cpp

```diff
@@ -180,6 +180,49 @@ const videoio_container_params_t videoio_container_params[] =
 
 INSTANTIATE_TEST_CASE_P(/**/, videoio_container, testing::ValuesIn(videoio_container_params));
 
+typedef tuple<VideoCaptureAPIs, string, string, string, int> videoio_skip_params_t;
+typedef testing::TestWithParam< videoio_skip_params_t > videoio_skip;
+
+TEST_P(videoio_skip, read)
+{
+    const VideoCaptureAPIs api = get<0>(GetParam());
+
+    if (!videoio_registry::hasBackend(api))
+        throw SkipTestException("Backend was not found");
+
+    const string path = get<1>(GetParam());
+    const string ext = get<2>(GetParam());
+    const string env = get<3>(GetParam());
+    const int expectedFrameNumber = get<4>(GetParam());
+
+    #ifdef _WIN32
+        _putenv_s("OPENCV_FFMPEG_CAPTURE_OPTIONS", env.c_str());
+    #else
+        setenv("OPENCV_FFMPEG_CAPTURE_OPTIONS", env.c_str(), 1);
+    #endif
+
+    VideoCapture container(findDataFile(path + "." + ext), api);
+    ASSERT_TRUE(container.isOpened());
+    Mat reference;
+    int nframes = 0, n_err = 0;
+    while (container.read(reference) && n_err < 3)
+        nframes++;
+    ASSERT_EQ(nframes, expectedFrameNumber);
+
+    #ifdef _WIN32
+        _putenv_s("OPENCV_FFMPEG_CAPTURE_OPTIONS", "");
+    #else
+        setenv("OPENCV_FFMPEG_CAPTURE_OPTIONS", "", 1);
+    #endif
+}
+
+const videoio_skip_params_t videoio_skip_params[] =
+{
+    videoio_skip_params_t(CAP_FFMPEG, "video/big_buck_bunny", "mp4", "avdiscard;nonkey", 11)
```

Comment:
The test is FFMPEG specific, so no need to push backend type as parameter. Also file name and extension are always used together, no need to pass them as separate parameters and then concatenate in test body.